tests/api-watch-diff/test.ml

```diff
@@ -213,3 +213,17 @@ let%expect_test "Modifying a module" =
   in
   Format.printf "%a" pp_diff_list result;
   [%expect {|[Any]|}]
+
+let%expect_test "Reference more general than Current interface" =
+  let reference = Test_helpers.compile_interface {|val x : 'a list|} in
+  let current = Test_helpers.compile_interface {|val x : float list|} in
+  let result = Api_watch_diff.diff_interface ~reference ~current in
+  Format.printf "%a" pp_diff_list result;
+  [%expect {|[Value (x, Modified)]|}]
+
+let%expect_test "Current more general than Reference interface" =
+  let reference = Test_helpers.compile_interface {|val x : int list|} in
+  let current = Test_helpers.compile_interface {|val x : 'a list|} in
+  let result = Api_watch_diff.diff_interface ~reference ~current in
+  Format.printf "%a" pp_diff_list result;
+  [%expect {|[Value (x, Modified)]|}]
```

Comment:
You could potentially do all of this in the same test here since we're trying to show that the modification is detected both ways. You can have multiple "expect statement" in the same test:
```suggestion
  [%expect {|[Value (x, Modified)]|}];
  let rev_result =
    Api_watch_diff.diff_interface ~reference:current ~current:reference
  in
  Format.printf "%a" pp_dif_list rev_result;
  [%expect {|[Value (x, Modified)]|}]
```

To clarify things we'd need to rename `reference` and `current` to `general` and `specialized` for instance, and change the name of the test to reflect that it's not focused on a single "direction".